streams/dshr_tinterp_mod.F90

```diff
@@ -238,27 +238,18 @@ subroutine shr_tInterp_getAvgCosz(tavCosz, lon, lat, &
 
     do while( reday < reday2) ! mid-interval t-steps thru interval [LB,UB]
 
+       !--- get next cosz value for t-avg ---
+       call shr_tInterp_getCosz(cosz,lon,lat,ymd,tod,eccen,mvelpp,lambm0,obliqr,calendar)
+       n = n + ldt
+       tavCosz = tavCosz + cosz*real(ldt,r8)  ! add to partial sum
+
        !--- advance to next time in [LB,UB] ---
        ymd0 = ymd
        tod0 = tod
        reday0 = reday
        call shr_cal_advDateInt(ldt,'seconds',ymd0,tod0,ymd,tod,calendar)
        call shr_cal_timeSet(reday,ymd,tod,calendar)
 
-       if (reday > reday2) then
```

Comment:
OK, looking at this, the reason the previous code is wrong is that it's essentially doing the interval (LB,UB] so not starting at  ldt after LB (Lower bound) (so LB is exclusive) and continuing until UB (Upper Bound) inclusive. So LB exclusive and UB inclusive. But, once you reach the UB, you go to the next interval. So it should be [LB,UB) LB inclusive, and UB exclusive, which is what the new code changes do.